homeassistant/components/glances/sensor.py

```diff
@@ -332,11 +332,12 @@ def __init__(
     def available(self) -> bool:
         """Set sensor unavailable when native value is invalid."""
         if super().available:
+            value = self.native_value
             return (
                 not self._numeric_state_expected
-                or isinstance(self.native_value, (int, float))
-                or isinstance(self.native_value, str)
-                and self.native_value.isnumeric()
+                or isinstance(value, (int, float))
```

Comment:
You could use a walrus here
```suggestion
                or isinstance(value := self.native_value, (int, float))
```

Reply:
I don't understand the purpose of this change. Please clarify..
The purpose of this PR is to remove additional calls to `self.native_value`, why should we explicitly call it again here? `value` is already assigned..

Reply:
OK it makes sense when line 335 is removed.
The code gets shorter, but concerning readability I am not sure if walrus is better...
What do you think? Would you still commend using walrus?